Zend/zend_builtin_functions.c

```diff
@@ -1273,9 +1270,19 @@ ZEND_FUNCTION(get_object_vars)
 						const char *prop_name, *class_name;
 						size_t prop_len;
 						zend_unmangle_property_name_ex(key, &class_name, &prop_name, &prop_len);
+						/* We assume here that a mangled property name is never
+						 * numeric. This is probably a safe assumption, but
+						 * theoretically someone might write an extension with
+						 * private, numeric properties. Well, too bad.
```

Comment:
I mean, in theory, but I don't see why anyone would actually write a class that defines properties like `->{'123'}`, it would just be awkward in use. After all, you can't do that in userland (you can set them dynamically, though).